kore/src/Kore/Log.hs

```diff
@@ -285,28 +282,22 @@ makeKoreLogger exeName timestampSwitch logToText =
     indent = Pretty.indent 4
 
 -- | Adds the current timestamp to a log entry.
-withTimestamp :: MonadIO io => ActualEntry -> io WithTimestamp
-withTimestamp msg = WithTimestamp msg <$> getLocalTime
-
--- Helper to get the local time in 'MonadIO'.
-getLocalTime :: MonadIO m => m LocalTime
-getLocalTime =
-    liftIO $ utcToLocalTime <$> getCurrentTimeZone <*> getCurrentTime
-
--- Formats the local time using the provided format string.
-formatLocalTime :: IsString s => String -> LocalTime -> s
-formatLocalTime format = fromString . formatTime defaultTimeLocale format
+withTimestamp :: MonadIO io => TimeSpec -> ActualEntry -> io WithTimestamp
+withTimestamp startTime msg = liftIO $ do
+    currentTime <- getTime Monotonic
+    pure $ WithTimestamp msg (diffTimeSpec currentTime startTime)
```

Comment:
Let's use the `currentTime` here, and calculate the difference when the log message is emitted. Then, we don't need to inject the `startTime` so deeply.

```suggestion
withTimestamp :: MonadIO io => ActualEntry -> io WithTimestamp
withTimestamp msg = liftIO $ do
    currentTime <- getTime Monotonic
    pure $ WithTimestamp msg currentTime
```